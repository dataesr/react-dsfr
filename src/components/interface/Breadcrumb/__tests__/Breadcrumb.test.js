import { render, screen } from '@testing-library/react';
import { Breadcrumb, BreadcrumbItem } from '..';

describe('<Breadcrumb />', () => {
  beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // Deprecated
        removeListener: jest.fn(), // Deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });
  it('should render breadcrumb properly', () => {
    render(
      <Breadcrumb>
        <BreadcrumbItem href="#">Home</BreadcrumbItem>
        <BreadcrumbItem href="#">Productions</BreadcrumbItem>
        <BreadcrumbItem current>
          Production 1234567899 avec un nom de page hyper long qui devrait etre tronqu√©
        </BreadcrumbItem>
      </Breadcrumb>,
    );
    const nav = screen.getByTestId('bc-nav');
    expect(nav.className).toBe('fr-breadcrumb');
    const bcList = screen.getByTestId('bc-list');
    expect(bcList.className).toBe('fr-breadcrumb__list');
    const button = screen.getByTestId('bc-button');
    expect(button.className).toBe('fr-breadcrumb__button');
    const items = screen.getAllByTestId('bc-item');
    expect(items).toHaveLength(3);
    expect(items[0].className).toBe('fr-breadcrumb__item');
    expect(items[2].className).toMatch(/fr-breadcrumb__item--current/);
    expect(nav).toMatchSnapshot();
  });
});
