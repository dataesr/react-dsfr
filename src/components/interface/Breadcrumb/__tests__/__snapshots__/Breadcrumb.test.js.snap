// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumb /> should render breadcrumb properly 1`] = `
<nav
  aria-label="Fil d'ariane"
  class="fr-breadcrumb"
  data-testid="breadcrumb"
  style="max-height: calc(0px + 1rem);"
>
  <button
    class="fr-breadcrumb__button"
    data-testid="bc-button"
    type="button"
  >
    Voir le fil d'ariane
  </button>
  <div
    class="fr-collapse"
  >
    <ol
      class="fr-breadcrumb__list"
      data-testid="bc-list"
      hidden=""
    >
      <li
        class="fr-breadcrumb__item"
        data-testid="bcitem"
      >
        <a
          class="fr-breadcrumb__link"
          href="#"
          target="_self"
        >
          Home
        </a>
      </li>
      <li
        class="fr-breadcrumb__item"
        data-testid="bcitem"
      >
        <a
          class="fr-breadcrumb__link"
          href="#"
          target="_self"
        >
          Productions
        </a>
      </li>
      <li
        class="fr-breadcrumb__item fr-breadcrumb__item--current"
        data-testid="bcitem"
      >
        <a
          aria-current="page"
          class="fr-breadcrumb__link"
          target="_self"
        >
          Production 1234567899 avec un nom de page hyper long qui devrait etre tronqu√©
        </a>
      </li>
    </ol>
  </div>
</nav>
`;
